frontend/src/components/history/HistoryAttemptModal.tsx

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import {
+  useDisclosure,
+  Button,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  ModalBody,
+  ModalFooter,
+  HStack,
+} from "@chakra-ui/react";
+import { Question } from "../../proto/types";
+import QuestionSection from "../question/QuestionSection";
+import HistorySection from "./HistorySection";
+import saveFile from "../../utils/fileDownloadUtil";
+import { Language } from "../../types";
+
+type Props = {
+  submission: string;
+  question: Question;
+  attemptId: number;
+  users: string[];
+  language: string;
+};
+
+function HistoryAttemptModal({
+  question,
+  submission,
+  attemptId,
+  users,
+  language,
+}: Props) {
+  const { isOpen, onOpen, onClose } = useDisclosure();
```

Comment:
This hook should exist outside of the modal itself, you can take reference from `pages/Session.tsx` to see how it is done. 

The objective is that the modal itself should be a "dumb" component.